app.py

```diff
@@ -76,18 +79,28 @@ def stats(api, memory_used_gauge, memory_total_gauge, network_up_gauge, network_
         disk_name_info.labels(disk_id, disk_name).info({"disk_name": disk_name})
 
         smart_status = str(api.storage.disk_smart_status(disk_id))
-        s_status_enum.labels(disk_id, disk_name).state(smart_status)
+        if smart_status not in smart_states:
+            smart_status = "error"
+
+        smart_status_enum.labels(disk_id, disk_name).state(smart_status)
+
+        disk_status = str(api.storage.disk_status(disk_id))
+        if disk_status not in disk_states:
+            disk_status = "error"
 
-        status = str(api.storage.disk_status(disk_id))
-        status_enum.labels(disk_id, disk_name).state(status)
+        disk_status_enum.labels(disk_id, disk_name).state(disk_status)
 
         disk_temp = api.storage.disk_temp(disk_id)
         disk_temp_gauge.labels(disk_id, disk_name).set(disk_temp)
 
     for share_id in api.share.shares_uuids:
-        share_name = str(api.share.share_name(share_id))
-        share_size_used = str(api.share.share_size(share_id, human_readable=False))
-        share_size_used_gauge.labels(share_id, share_name).set(share_size_used)
+        try:
+            share_name = str(api.share.share_name(share_id))
+            share_size_used = str(api.share.share_size(share_id, human_readable=False))
+            share_size_used_gauge.labels(share_id, share_name).set(share_size_used)
+        # no metrics available, which could be the case with unsupported external devices
```

Comment:
What does "unsupported external devices" mean in this context?

Reply:
The Synology API (and thus the web backend) doesn't list share sizes for external drives which are plugged in via USB. Without the try block, the exporter fail the moment I plug in a hard drive. The only relevant metric for the shares are the share size, so without it, we can simply ignore the external drive share and move on to the next. I didn't feel like this had to be logged or otherwise cached.